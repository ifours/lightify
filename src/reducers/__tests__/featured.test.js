import reducer, { initialState } from '../featured';

describe('Featured reducer', () => {
  it('should return initialState', () => {
    const state = reducer(undefined, {});

    expect(state).toMatchSnapshot();
    expect(state).toBe(initialState);
  });

  it('should return new state for FETCH_FEATURED_PLAYLISTS_SUCCEEDED action', () => {
    const result = {
      message: 'foo',
      playlists: {
        'bar': 'baz',
        items: [],
      },
    };
    const state = reducer(undefined, { type: 'FETCH_FEATURED_PLAYLISTS_SUCCEEDED', payload: { result } });

    expect(state).toMatchSnapshot();
  });
});
