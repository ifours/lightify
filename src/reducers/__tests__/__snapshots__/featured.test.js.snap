// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Featured reducer should return initialState 1`] = `
Object {
  "message": "",
  "playlists": Object {
    "items": Array [],
  },
}
`;

exports[`Featured reducer should return new state for FETCH_FEATURED_PLAYLISTS_SUCCEEDED action 1`] = `
Object {
  "message": "foo",
  "playlists": Object {
    "bar": "baz",
    "items": Array [],
  },
}
`;
